### PR TITLE
Add more options to config_template for docker image

### PR DESCRIPTION
**Pull request by peaaceChoi**
**What changed?**
for #3870 
* Add more options to config_template for docker image
  - FRONTEND_PORT
  - MATCHING_PORT
  - WORKER_PORT
  - HISTORY_PORT
  - VISIBILITY_NAME
  - ES_PORT

**Why?**
It is more convenient if a config_template is provided dynamically. In my case, some ports were already assigned to other services.

**How did you test it?**
Local test 

**Potential risks**
It doesn't matter as it supports default values.

### CODE REVIEW

**Review by meiliang86 — APPROVED** [review]

**Review by mkolodezny — APPROVED** [review]

File: docker/config_template.yaml [hunk 1]
```diff
@@ -252,19 +252,19 @@ kafka:
...
     applications:
         visibility:
-            topic: cadence-visibility-dev
-            dlq-topic: cadence-visibility-dev-dlq
+            topic: {{ default .Env.VISIBILITY_NAME "cadence-visibility-dev" }}
+            dlq-topic: {{ default .Env.VISIBILITY_NAME "cadence-visibility-dev" }}-dlq
 
 publicClient:
-    hostPort: {{ default .Env.BIND_ON_IP "127.0.0.1" }}:7933
+    hostPort: {{ default .Env.BIND_ON_IP "127.0.0.1" }}:{{ default .Env.FRONTEND_PORT "7933" }}
```
Comment (longquanzheng) [inline docker/config_template.yaml:267]: `BIND_ON_IP` won't work(and it's already broken here for some optional feature like XDC/batcher).
It should be frontend service address, like https://github.com/banzaicloud/banzai-charts/blob/cd719b684baae9e0d32785435205710f8ee32682/cadence/templates/server-configmap.yaml#L163

 Can you help change this to a new option? like `.Env.FRONTEND_SERVICE`

File: docker/config_template.yaml [hunk 2]
```diff
@@ -207,13 +207,13 @@ clusterMetadata:
             enabled: true
             initialFailoverVersion: 0
             rpcName: "cadence-frontend"
-            rpcAddress: {{ default .Env.PRIMARY_SEEDS "cadence" }}:7933
+            rpcAddress: {{ default .Env.PRIMARY_SEEDS "cadence" }}:{{ default .Env.FRONTEND_PORT "7933" }}
```
Comment (longquanzheng) [inline docker/config_template.yaml:210]: I think we should
`PRIMARY_SEEDS`=>`PRIMARY_FRONTEND_SERVICE`

File: docker/config_template.yaml [hunk 3]
```diff
@@ -207,13 +207,13 @@ clusterMetadata:
...
         {{- if .Env.ENABLE_GLOBAL_DOMAIN }}
         secondary:
             enabled: true
             initialFailoverVersion: 2
             rpcName: "cadence-frontend"
-            rpcAddress: {{ default .Env.SECONDARY_SEEDS "cadence-secondary" }}:7933
+            rpcAddress: {{ default .Env.SECONDARY_SEEDS "cadence-secondary" }}:{{ default .Env.FRONTEND_PORT "7933" }}
```
Comment (longquanzheng) [inline docker/config_template.yaml:216]: `SECONDARY_SEEDS`=>`SECONDARY_FRONTEND_SERVICE`

**Review by longquanzheng — CHANGES_REQUESTED** [review]
cc @Shaddoll @yux0

File: docker/config_template.yaml (hunk 1 above)
Comment (peaaceChoi) [reply docker/config_template.yaml]: Sure, I fixed and updated other `.Env.BIND_ON_IP` also.
plz, check new commit.

File: docker/config_template.yaml [hunk 4]
```diff
@@ -101,18 +101,18 @@ ringpop:
...
     {{- end }}
     maxJoinDuration: 30s
 
 services:
     frontend:
         rpc:
-            port: 7933
-            bindOnIP: {{ default .Env.BIND_ON_IP "127.0.0.1" }}
+            port: {{ default .Env.FRONTEND_PORT "7933" }}
+            bindOnIP: {{ default .Env.FRONTEND_SERVICE "127.0.0.1" }}
```
Comment (longquanzheng) [inline docker/config_template.yaml:115]: ooops, you misunderstood my comment. I didn't mean to change this. `BIND_ON_IP` is correct here.

File: docker/config_template.yaml [hunk 5]
```diff
@@ -132,8 +132,8 @@ services:
 
     matching:
         rpc:
-            port: 7935
-            bindOnIP: {{ default .Env.BIND_ON_IP "127.0.0.1" }}
+            port: {{ default .Env.MATCHING_PORT "7935" }}
+            bindOnIP: {{ default .Env.MATCHING_SERVICE "127.0.0.1" }}
```
Comment (longquanzheng) [inline docker/config_template.yaml:136]: same

File: docker/config_template.yaml [hunk 6]
```diff
@@ -153,8 +153,8 @@ services:
 
     history:
         rpc:
-            port: 7934
-            bindOnIP: {{ default .Env.BIND_ON_IP "127.0.0.1" }}
+            port: {{ default .Env.HISTORY_PORT "7934" }}
+            bindOnIP: {{ default .Env.HISTORY_SERVICE "127.0.0.1" }}
```
Comment (longquanzheng) [inline docker/config_template.yaml:157]: same

File: docker/config_template.yaml [hunk 7]
```diff
@@ -174,8 +174,8 @@ services:
 
     worker:
         rpc:
-            port: 7939
-            bindOnIP: {{ default .Env.BIND_ON_IP "127.0.0.1" }}
+            port: {{ default .Env.WORKER_PORT "7939" }}
+            bindOnIP: {{ default .Env.WORKER_SERVICE "127.0.0.1" }}
```
Comment (longquanzheng) [inline docker/config_template.yaml:178]: same

**Review by longquanzheng — CHANGES_REQUESTED** [review]
Sorry, what I meant is to change the `BIND_ON_IP` in `publicClient`, not others

File: docker/config_template.yaml (hunk 1 above)
Comment (longquanzheng) [reply docker/config_template.yaml]: >  updated other .Env.BIND_ON_IP also.

This is not correct. don't update others. 
Sorry it's a bit complicated. 
`BIND_ON_IP` is correct for `ringpop` configuration. While  in `publicClient` and `clusterInfomation`, we need to put the address of Frontend service.

**Review by longquanzheng — APPROVED** [review]

**Comment by peaaceChoi** [comment]
@longquanzheng I understood the details of `bind-ip`, Thanks for your review. :)

**Comment by longquanzheng** [comment]
@peaaceChoi can you sign the CLA? thanks

**Comment by longquanzheng** [comment]
> @longquanzheng I understood the details of `bind-ip`, Thanks for your review. :)

Thanks for your contribution. But the CLA is required to merge.

**Comment by peaaceChoi** [comment]
@longquanzheng  sure,  I approved it.

**Pull request merged**